src/app/modules/details/test-exec-details.component.ts

```diff
@@ -89,22 +92,41 @@ export class TestExecDetailsComponent implements OnInit, OnDestroy {
               this.rawLog = entry.content;
             }
             break;
-      }});
+        }
+      });
+      this.imagesRemainingToLoad = screenshotPaths.length;
+      this.encodedScreenshots = [...Array(screenshotPaths.length)];
+      screenshotPaths.forEach((path, index) => this.getScreenshot(path, index));
     } else {
       console.log('warning: received empty details data');
     }
   }
 
-  private createImageFromBlob(image: Blob) {
+  private getScreenshot(path: string, index: number) {
+    this.showImages = false;
+    this.resourceService.getBinaryResource(path).then((screenshot: Blob) => {
+      console.log(`retrieved screenshot ${screenshot}`);
+      this.createImageFromBlob(screenshot, index);
+    }, error => {
+      console.log(error);
+      this.showImages = false;
+    });
+  }
+
+  private createImageFromBlob(image: Blob, index: number) {
     const reader = this.fileReaderProvider.get();
     reader.onload = () => {
-       this.encodedScreenshot = reader.result;
-       console.log('screenshot was set');
+      this.encodedScreenshots[index] = reader.result;
+      this.imagesRemainingToLoad--;
+      console.log('screenshot was added');
```

Comment:
for the root node this might log a lot of screen shots, doesn't it? guess this log line is not needed anymore ..